scripts/AbinsModules/CalculatePowder.py

```diff
@@ -0,0 +1,116 @@
+    """
+    Class for calculating powder data.
+    """
+    def __init__(self, filename=None, abins_data=None):
+        """
+        @param filename:  name of input DFT filename
+        @param abins_data: object of type AbinsData with data from input DFT file
+        """
+
+        if not isinstance(abins_data, AbinsData):
+            raise ValueError("Object of AbinsData was expected.")
+        self._abins_data = abins_data
+
+        super(CalculatePowder, self).__init__(input_filename=filename,
+                                              group_name=AbinsParameters.powder_data_group + "/")
+
+    def _get_gamma_data(self, k_data=None):
+        """
+        Extracts k points data only for Gamma point.
+        @param k_data: numpy array with k-points data.
+        @return: dictionary with k_data only for Gamma point
+        """
+        gamma_pkt_index = -1
+
+        # look for index of Gamma point
+        num_k = k_data["k_vectors"].shape[0]
+        for k in range(num_k):
+            if np.linalg.norm(k_data["k_vectors"][k]) < AbinsConstants.SMALL_K:
+                gamma_pkt_index = k
+                break
+        if gamma_pkt_index == -1:
+            raise ValueError("Gamma point not found.")
+
+        k_points = {"weights": k_data["weights"][gamma_pkt_index],
+                    "k_vectors": k_data["k_vectors"][gamma_pkt_index],
+                    "frequencies": k_data["frequencies"][gamma_pkt_index],
+                    "atomic_displacements": k_data["atomic_displacements"][gamma_pkt_index]}
+
+        return k_points
+
+    def _calculate_powder(self):
+        """
+        Calculates powder data (a_tensors, b_tensors according to aCLIMAX manual).
+        """
+
+        # get all necessary data
+        data = self._abins_data.extract()
+        k_data = self._get_gamma_data(data["k_points_data"])
+        displacements = k_data["atomic_displacements"]
+        num_atoms = displacements.shape[0]
+        atoms_data = data["atoms_data"]
+        freq_hartree = k_data["frequencies"][AbinsConstants.FIRST_OPTICAL_PHONON:]  # frequencies in Hartree units
+
+        frequencies = freq_hartree / AbinsConstants.CM1_2_HARTREE  # convert frequencies to cm^1
+
+        powder = PowderData(num_atoms=num_atoms)
+
+        # Notation for  indices:
+        #     num_freq -- number of optical phonons (total number of phonons - acoustic phonons)
+        #     num_atoms -- number of atoms
+        #     dim -- size of displacement vector for one atom (dim = 3)
+
+        # masses[num_atoms, num_freq]
+        masses = np.asarray([([atom["mass"]] * frequencies.size) for atom in atoms_data])
+        masses = masses / AbinsConstants.M_2_HARTREE  # convert  mass of atoms to amu units
+
+        # disp[num_atoms, num_freq, dim]
+        disp = displacements[:, AbinsConstants.FIRST_OPTICAL_PHONON:]
+
+        # factor[num_atoms, num_freq]
+        factor = np.einsum('ij,j->ij', masses, AbinsConstants.ACLIMAX_CONSTANT / frequencies)
```

Comment:
Yes, unfortunately rhel6 does not support np.einsum, and you cannot turn off the build for rhel6. If you only used np.einsum in a specific test or some very specialised functionality you could deable that function in the code, but to me it looks like you are here using np.einsum in core functionally in your code. 

Also tried to see if there was a scipy equivalent method, but could not find one. Hence could you look into finding another way of doing this summation